### PR TITLE
Test step flaky tests should show up to the left of the dashboard

### PR DESCRIPTION
**Pull request by muditatandon**
Test step flaky tests should show up to the left of the dashboard.

Bug: flutter/flutter#102321

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by CaseyHillers** [comment]
You can use https://github.com/flutter/cocoon/blob/main/format.sh for code formatting in Cocoon (more generally every repo has some format.sh to make this easier except flutter/flutter)

File: dashboard/lib/widgets/task_grid.dart [hunk 1]
```diff
@@ -214,7 +223,7 @@ class _TaskGridState extends State<TaskGrid> {
         }
         if (commitCount <= 25) {
           String weightStatus = task.status;
-          if (task.isFlaky) {
```
Comment (CaseyHillers) [inline dashboard/lib/widgets/task_grid.dart:217]:
```suggestion
          if (task.isFlaky || task.isTestFlaky) {
```

**Comment by muditatandon** [comment]
> You can use https://github.com/flutter/cocoon/blob/main/format.sh for code formatting in Cocoon (more generally every repo has some format.sh to make this easier except flutter/flutter)

**Comment by keyonghan** [comment]
Can you try `flutter test --update-goldens` on a linux bot to fix the golden errors?

**Review by keyonghan — COMMENTED** [review]
can we add a unit test?

File: dashboard/test/widgets/task_grid_test.dart [hunk 2]
```diff
@@ -581,6 +581,45 @@ void main() {
...
+              authService: MockGoogleSignInService(),
+              cocoonService: MockCocoonService(),
+            ),
+            commitStatuses: <CommitStatus>[
+              CommitStatus()
+                ..commit = (Commit()..author = 'Cast')
+                ..tasks.addAll(
+                  <Task>[
+                    Task()
+                      ..stageName = 'A'
```
Comment (CaseyHillers) [inline dashboard/test/widgets/task_grid_test.dart:599]: Nit: we should move these to name instead of stageName

**Comment by ricardoamador** [comment]
Reopening as I have made changes but cannot pull down this PR. Opening from my fork.

**Pull request closed without merging**